Gems/LmbrCentral/Code/Source/Bundling/BundlingSystemComponent.cpp

```diff
@@ -114,15 +126,130 @@ namespace LmbrCentral
         m_bundleModeBundles.clear();
     }
 
+#if defined(CARBONATED)
+    /**
+     * Loads a specific bundle, optionally reloading it if it has been previously opened.
+     * There's a good amount of copy-pasta here.  Intentional, to reduce refactor-deltas in LY vs Carbonated code.
+     */
+    void BundlingSystemComponent::LoadBundle(const char* szPath, bool reload)
+    {
+        auto archive = AZ::Interface<AZ::IO::IArchive>::Get();
+        if (!archive)
+        {
+            AZ_Error("BundlingSystem", false, "Couldn't Get IArchive to load bundles!");
+            return;
+        }
+
+        AZStd::lock_guard<AZStd::mutex> openBundleLock(m_bundleModeMutex);
+
+        // NOTE:  'bundlePath' must produce an identical result to paks discovered below in "LoadBundles"
+        AZStd::string bundlePath;
+        AZ::StringFunc::Path::Join(bundleRoot, szPath, bundlePath);
+
+        bool pakOpened = false;
+        for (const auto& openedBundle : m_bundleModeBundles)
+        {
+            if (openedBundle == bundlePath)
+            {
+                pakOpened = true;
+                break;
+            }
+        }
+
+        if (pakOpened)
+        {
+            if (reload)
+            {
+                if (archive->IsPackOpen(AZStd::string_view(szPath)))
+                {
+                    if (!archive->ClosePack(szPath))
+                    {
+                        AZ_Warning("BundlingSystem", false, "Failed to close bundle %s", szPath);
+                    }
+                }
+
+                m_bundleModeBundles.erase(AZStd::remove(m_bundleModeBundles.begin(), m_bundleModeBundles.end(), bundlePath.c_str()), m_bundleModeBundles.end());
+            }
+            else
+            {
+                return;
```

Comment:
Quick return is a better option, like
if (!reload)
{
    return;
}
... reload code...
The logic is more clear, the the major code block indent level is less than if/else case,

Reply:
Updated